### DISCUSSION TITLE
Update giflib formula to version 5.2.2

### DISCUSSION THREAD

**Discussion by BobbyVillaluz** [Homebrew/discussions#5261] [Writing Formulae/Casks]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

[giflib 5.2.2](https://sourceforge.net/p/giflib/code/ci/master/tree/NEWS) was released on 2/19/2024. The PR https://github.com/Homebrew/homebrew-core/pull/163204 was submitted to update the Homebrew formula, but it appears to have been closed due to an upstream issue with giflib's `doc` [Makefile](https://sourceforge.net/p/giflib/code/ci/master/tree/doc/Makefile) relying on the ImageMagick `convert` tool (which may not be installed in every environment).

Is it feasible to update the [formula](https://github.com/miccal/homebrew-core/blob/master/Formula/g/giflib.rb) to 5.2.2? For example, could a conditional ImageMagick dependency be included (since giflib [only builds its doc package for non-Darwin operating systems](https://sourceforge.net/p/giflib/code/ci/5.2.2/tree/Makefile#l90))?
```
depends_on "imagemagick" unless OS.mac?
```

Or do we have to wait for the next giflib release? @chenrui333 [mentioned](https://github.com/Homebrew/homebrew-core/pull/163204#issuecomment-1952712209) discussing the removal of this build step with @eric-s-raymond.

**Comment by SMillerDev** [comment]
If someone can make the update work we can merge it, so a pull request would be much appreciated

**Reply by BobbyVillaluz** [reply]
@SMillerDev I got the update to work (passed all tests locally) in [#168690](https://github.com/Homebrew/homebrew-core/pull/168690) but it looks like some of the GitHub CI checks are timing out after 60 minutes, since there are so many formulae dependent on giflib.

**Reply by SMillerDev** [reply]
That number doesn't seem to exist and the link doesn't go anywhere, are you sure it's correct?

**Reply by BobbyVillaluz** [reply]
Sorry about that. The link should work now
